examples/expressions/12_subsampling.py

```diff
@@ -0,0 +1,89 @@
+"""
+.. currentmodule:: skrub
+
+.. _example_subsampling:
+
+Subsampling for faster development
+==================================
+
+Here we show how to use :meth:`.skb.subsample()
+<Expr.skb.subsample>` to speed-up interactive creation of skrub
+expressions by subsampling the data when computing preview results.
+"""
+
+# %%
+
+import skrub
+import skrub.datasets
+
+dataset = skrub.datasets.fetch_employee_salaries().employee_salaries
+
+full_data = skrub.var("data", dataset)
+full_data
+
+# %%
+# We are working with a dataset of over 9K rows. As we build up our pipeline,
+# we see previews of the intermediate results so we can check that it behaves
+# as we expect. However, if some estimators are slow, fitting them and
+# computing results on the whole data can slow us down.
+#
+# So we can tell skrub to subsample the data when computing the previews, with
+# :meth:`.skb.subsample() <Expr.skb.subsample>`.
+
+# %%
+data = full_data.skb.subsample(n=100)
+data
+
+# %%
+# The rest of the pipeline will now use only 100 points for its previews.
+#
+# .. note::
+#
+#    By default subsampling is applied *only for previews*: the results
+#    shown when we display the expression, and the output of calling
+#    :meth:`.skb.preview() <Expr.skb.preview>`. For other methods such as
+#    :meth:`.skb.get_pipeline() <Expr.skb.get_pipeline>` or
+#    :meth:`.skb.cross_validate() <Expr.skb.cross_validate>`, *no subsampling is
+#    done by default*. We can explicitly ask for it with ``keep_subsampling=True``
+#    as we will see below.
+#
+# To continue our pipeline we now define X and y:
+
+# %%
+employees = data.drop(columns="current_annual_salary", errors="ignore").skb.mark_as_X()
```

Comment:
That line is a bit long. It overflows in the website layout.

I guess we can't do much about this, because the code formatters like it this why.